### PR TITLE
initial commit of selinux

### LINKED ISSUE #85: Add SELinux policies

### PR DESCRIPTION
**Pull request by cooktheryan**
**Which issue(s) this PR addresses**:
Closes #85 

This should only be a placeholder until we get into https://github.com/containers/container-selinux/blob/main/container.fc

That should be the next goal from there it would be important to see how long it would be until we get into the specific OS channels.

Also, @copejon we may want to ship the microshift.pp file rather than generate one but I wanted to have this up for @sallyom to review

### CODE REVIEW

**Comment by cooktheryan** [comment]
```
[vagrant@fedora ~]$ kubectl get po -wA
NAMESPACE                       NAME                                  READY   STATUS    RESTARTS   AGE
default                         wordpress-mysql-6c479567b-gjs2g       1/1     Running   0          171m
kube-system                     kube-flannel-ds-7l58f                 1/1     Running   0          175m
kubevirt-hostpath-provisioner   kubevirt-hostpath-provisioner-fhjr6   1/1     Running   0          175m
openshift-dns                   dns-default-gfsw7                     3/3     Running   0          175m
openshift-ingress               router-default-6d8c9d8f57-b2frf       1/1     Running   0          175m
openshift-service-ca            service-ca-64547678c6-w7xvv           1/1     Running   0          175m
^C[vagrant@fedora ~]$ getenforce
Enforcing

File: install.sh [hunk 1]
```diff
@@ -52,14 +52,12 @@ register_subs() {
...
-    #   sudo semanage fcontext -m -t container_runtime_exec_t /usr/local/bin/microshift
-    # sudo mkdir -p /var/lib/kubelet/
-    # sudo chcon -R -t container_file_t /var/lib/kubelet/
-    # sudo chcon -R system_u:object_r:bin_t:s0 /usr/local/bin/microshift
-    sudo setenforce 0
-    sudo sed -i 's/SELINUX=enforcing/SELINUX=disabled/g' /etc/selinux/config
+build_selinux_policy() {
+    ## Workaround until packaged as RPM
+    sudo dnf -y install selinux-policy-devel
+    curl -L -o /tmp/microshift.te https://gist.githubusercontent.com/cooktheryan/d5503ac73848e8f33554fba1f58eb021/raw/a9200ef29ed6b83638ef597d58204736c9af746f/gistfile1.txt
```
Comment (rootfs) [inline install.sh:58]: leave the changes for now after the selinux policy files are merged

**Comment by rootfs** [comment]
> ```
> [vagrant@fedora ~]$ kubectl get po -wA
> NAMESPACE                       NAME                                  READY   STATUS    RESTARTS   AGE
> default                         wordpress-mysql-6c479567b-gjs2g       1/1     Running   0          171m
> kube-system                     kube-flannel-ds-7l58f                 1/1     Running   0          175m
> kubevirt-hostpath-provisioner   kubevirt-hostpath-provisioner-fhjr6   1/1     Running   0          175m
> openshift-dns                   dns-default-gfsw7                     3/3     Running   0          175m
> openshift-ingress               router-default-6d8c9d8f57-b2frf       1/1     Running   0          175m
> openshift-service-ca            service-ca-64547678c6-w7xvv           1/1     Running   0          175m
> ^C[vagrant@fedora ~]$ getenforce
> Enforcing
> ```

does it survive a reboot?

**Comment by cooktheryan** [comment]
@rootfs survives reboot
```
[fedora@ip-172-31-88-225 ~]$ kubectl get po -A
NAMESPACE                       NAME                                  READY   STATUS    RESTARTS   AGE
kube-system                     kube-flannel-ds-rgtv5                 1/1     Running   0          7m29s
kubevirt-hostpath-provisioner   kubevirt-hostpath-provisioner-d45ff   1/1     Running   0          7m29s
openshift-dns                   dns-default-q9jpk                     3/3     Running   0          7m29s
openshift-ingress               router-default-6d8c9d8f57-sqtph       1/1     Running   1          7m29s
openshift-service-ca            service-ca-64547678c6-wzb27           1/1     Running   0          7m30s
[fedora@ip-172-31-88-225 ~]$ uptime
 14:31:09 up 2 min,  2 users,  load average: 0.99, 0.47, 0.18
```

**Comment by rootfs** [comment]
@cooktheryan let's first check in the policies, and a follow up PR to update install.sh

**Comment by cooktheryan** [comment]
@rootfs done. Once merged I will open the second PR to use the committed files

**Pull request merged**